src/NServiceBus.Core/Transports/Msmq/ReceiveWithNativeTransaction.cs

```diff
@@ -20,6 +20,11 @@ class ReceiveWithNativeTransaction : ReceiveStrategy
 
                     var message = inputQueue.Receive(TimeSpan.FromMilliseconds(10), msmqTransaction);
 
+                    if (message == null)
```

Comment:
I thought we discussed the null check is not necessary. At least you tried to convince me about that. If that is still true why was id added here?